### PR TITLE
Ferc1 rate base tag updates

### LINKED ISSUE #3479: Review Rate Base FERC1 Table
Okay @jrea-rmi ! Once we merge in PR #3450, I believe we now have much cleaner detailed tables and rate base table. Can you give these a review as best you can and let me know if you find anything that seems like a red flag? I've attached the two core graphs below and here are pickled dfs of the main inputs and the rate base table itself:
- [out_ferc1__yearly_rate_base.pkl.zip](https://github.com/catalyst-cooperative/pudl/files/14650786/out_ferc1__yearly_rate_base.pkl.zip)
- [out_ferc1__detailed_balance_sheet_assets.pkl.zip](https://github.com/catalyst-cooperative/pudl/files/14650783/out_ferc1__detailed_balance_sheet_assets.pkl.zip)
- [out_ferc1__detailed_balance_sheet_liabilities.pkl.zip](https://github.com/catalyst-cooperative/pudl/files/14650784/out_ferc1__detailed_balance_sheet_liabilities.pkl.zip)
- [out_ferc1__detailed_tags.pkl.zip](https://github.com/catalyst-cooperative/pudl/files/14650785/out_ferc1__detailed_tags.pkl.zip)
- [forest_as_table_core_ferc1__yearly_balance_sheet_assets_sched110.pkl.zip](https://github.com/catalyst-cooperative/pudl/files/14650777/forest_as_table_core_ferc1__yearly_balance_sheet_assets_sched110.pkl.zip)
- [forest_as_table_core_ferc1__yearly_income_statements_sched114.pkl.zip](https://github.com/catalyst-cooperative/pudl/files/14650778/forest_as_table_core_ferc1__yearly_income_statements_sched114.pkl.zip)

image
image

### Rate base table: by input table & utility
image
image
image
image
image
image

I'd be happy to give you a tour of these or set you up with anything you need to best get a sense of these tables. Even high level gut reactions are vert helpful!

Once we all feel good about the contents we can move on [#3310](https://github.com/catalyst-cooperative/pudl/issues/3310)

## Compiled Found Weird Items:
 - 2004: both the assets and liabilities table are abnormally low that year.

### PR DESCRIPTION
**Pull request by cmgosnell**
# Overview

Closes #3479.

What problem does this address?
I'm running off of the [check-list here](https://github.com/catalyst-cooperative/pudl/issues/3479#issuecomment-2020606527)

What did you change?

# Testing

How did you make sure this worked? How can a reviewer verify this?

```[tasklist]
# To-do list
- [ ] If updating analyses or data processing functions: make sure to update or write data validation tests (e.g.,  `test_minmax_rows()`)
- [ ] Update the [release notes](../docs/release_notes.rst): reference the PR and related issues.
- [ ] Ensure docs build, unit & integration tests, and test coverage pass locally with `make pytest-coverage` (otherwise the merge queue may reject your PR)
- [ ] Review the PR yourself and call out any questions or issues you have
- [ ] For minor ETL changes or data additions, once `make pytest-coverage` passes, make sure you have a fresh full PUDL DB downloaded locally, materialize new/changed assets and all their downstream assets and [run relevant data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation) using `pytest` and `--live-dbs`.
- [ ] For significant ETL, data coverage or analysis changes, once `make pytest-coverage` passes, ensure the full ETL runs locally and [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation) using `make pytest-validate` (a ~10 hour run). If you can't run this locally, run the `build-deploy-pudl` GitHub Action (or ask someone with permissions to). Then, check the logs on the `#pudl-deployments` Slack channel or `gs://builds.catalyst.coop`.
```

### CODE REVIEW

File: src/pudl/package_data/ferc1/xbrl_factoid_revenue_requirement_tags.csv
Comment (cmgosnell) [inline src/pudl/package_data/ferc1/xbrl_factoid_revenue_requirement_tags.csv:1]: this whole table had the old table names so I converted them all. I think this was one of the last set of tags we added when we were just about done with the *great rename* so I assume I missed this guy.

**Review by e-belfer — CHANGES_REQUESTED** [review]
Re: the GitHub commits, I'd try rebasing off main and cherry picking just the commits relevant to this branch. 

```
git checkout main
git pull
git checkout ferc1-rate-base-tag-updates
git rebase -i main # this should pull up a little interactive thingy that lets you only include *your* commits in your branch.
```
@jdangerx is a wizard at this and can probably give advice if you get stuck.

Regarding the actual tags, all ran well. A few questions:
- should `core_ferc1__yearly_balance_sheet_liabilities...` also have `net_ADIT` as the rate base category? I just see it on the assets side.
- For me, `other_regulatory_assets` still has `other_deferred_debits_and_credits` as the rate base tag - do you see the same?

**Comment by cmgosnell** [comment]
hm @jdangerx I'm clearly doing rebase wrong

```python
(pudl-dev) [pudl] git pull                                                                                                                                                                          16:51:03  ☁  main ☀
Already up to date.
(pudl-dev) [pudl] git checkout ferc1-rate-base-tag-updates                                                                                                                                          16:51:07  ☁  main ☀
Switched to branch 'ferc1-rate-base-tag-updates'
Your branch is up to date with 'origin/ferc1-rate-base-tag-updates'.
(pudl-dev) [pudl] git rebase -i main                                                                                                                                         16:51:16  ☁  ferc1-rate-base-tag-updates ☀
warning: skipped previously applied commit fce80ea8c
Successfully rebased and updated refs/heads/ferc1-rate-base-tag-updates.
(pudl-dev) [pudl] git push                                                                                                                                               16:51:59  ☁  ferc1-rate-base-tag-updates ☀ 𝝙 𝝙
To github.com:catalyst-cooperative/pudl.git
 ! [rejected]            ferc1-rate-base-tag-updates -> ferc1-rate-base-tag-updates (non-fast-forward)
error: failed to push some refs to 'github.com:catalyst-cooperative/pudl.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. If you want to integrate the remote changes,
hint: use 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
[1]    82169 exit 1     git push
(pudl-dev) [pudl] git pull                                                                                                                                               16:52:05  ☁  ferc1-rate-base-tag-updates ☀ 𝝙 𝝙
warning: skipped previously applied commit 796fe2978
warning: skipped previously applied commit dd895b727
warning: skipped previously applied commit d42b2a230
warning: skipped previously applied commit bb5bc98ec
Successfully rebased and updated refs/heads/ferc1-rate-base-tag-updates.
(pudl-dev) [pudl] git push                                                                                                                                                   16:52:26  ☁  ferc1-rate-base-tag-updates ☀
Everything up-to-date
(pudl-dev) [pudl]
```

**Comment by cmgosnell** [comment]
> * should `core_ferc1__yearly_balance_sheet_liabilities`... also have `net_ADIT` as the rate base category? I just see it on the assets side.

content wise this is kinda more of a jon question but there are facts labeled as `net_ADIT` on the liability side. there are three and they all start with `accumulated_deferred_income_taxes`

> *  For me, other_regulatory_assets still has other_deferred_debits_and_credits as the rate base tag - do you see the same?

yes! idk how i missed that one. updating now

**Comment by jdangerx** [comment]
Looks like you did everything right up through here - though I suppose you could have chosen the wrong commits in the `git rebase -i` editor:

```
(pudl-dev) [pudl] git pull                                                                                                                                                                          16:51:03  ☁  main ☀
Already up to date.
(pudl-dev) [pudl] git checkout ferc1-rate-base-tag-updates                                                                                                                                          16:51:07  ☁  main ☀
Switched to branch 'ferc1-rate-base-tag-updates'
Your branch is up to date with 'origin/ferc1-rate-base-tag-updates'.
(pudl-dev) [pudl] git rebase -i main                                                                                                                                         16:51:16  ☁  ferc1-rate-base-tag-updates ☀
warning: skipped previously applied commit fce80ea8c
Successfully rebased and updated refs/heads/ferc1-rate-base-tag-updates.
(pudl-dev) [pudl] git push                                                                                                                                               16:51:59  ☁  ferc1-rate-base-tag-updates ☀ 𝝙 𝝙
To github.com:catalyst-cooperative/pudl.git
 ! [rejected]            ferc1-rate-base-tag-updates -> ferc1-rate-base-tag-updates (non-fast-forward)
error: failed to push some refs to 'github.com:catalyst-cooperative/pudl.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. If you want to integrate the remote changes,
hint: use 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
[1]    82169 exit 1     git push
```

And it says "If you want to integrate the remote changes, use 'git pull'". But... you actually *don't* want to incorporate remote changes. So you want to then run `git push --force-with-lease` instead of  `git pull` and `git push`. See if that fixes your problem! Or we can hop on a call and I can walk you through it.

Details: by rebasing, you are making a bunch of new commits that have the same *content* as the old commits, so naturally the remote and local branches will look different. Git has no way to know if those commits on the remote branch, which are no longer on your remote branch, are "new changes that need to get pulled in" or "just the same old stuff we've just re-committed," which is why it prompts you to pull the new changes. Since you know it's just the same old stuff, you want to *overwrite* the history on the remote branch, which uses `git push --force`. `git push --force-with-lease` is a safer version of `git push --force`, which basically first checks that the remote branch hasn't changed since the last time you looked at it, before then applying a force push. It protects you from overwriting a commit on the remote that you don't even know about.

**Comment by cmgosnell** [comment]
yay @jdangerx  ty that totally worked :-) 🪄

**Review by e-belfer — APPROVED** [review]
>         should core_ferc1__yearly_balance_sheet_liabilities... also have net_ADIT as the rate base category? I just see it on the assets side.
> 
> content wise this is kinda more of a jon question but there are facts labeled as net_ADIT on the liability side. there are three and they all start with accumulated_deferred_income_taxes

I'm still seeing that `accumulated_deferred_income_taxes` is labelled as `net_ADIT` on the assets side and `<NA>` on the liabilities side. Is this expected behavior?

Otherwise everything works as expected and have verified all the checked off boxes in your list appear in the rate base table. Once confirmed that the above is expected, go ahead and merge it.

**Comment by cmgosnell** [comment]
> I'm still seeing that accumulated_deferred_income_taxes is labelled as net_ADIT on the assets side and <NA> on the liabilities side. Is this expected behavior?

oof yea ty for continuing to hone in on this. This was indeed an accident - a product of at first not realizing these was two `accumulated_deferred_income_taxes` records in the tag file. I think I assumed that the `accumulated_deferred_income_taxes_X` records on the liabilities side were the leafy records on the liabilities side. but that was an incorrect assumption idk why I didn't verify that. so ty good catch and its updated

**Pull request merged**